plugin/checker/check_helpers.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package main
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"go/types"
+
+	"github.com/mattermost/mattermost-server/plugin/checker/internal/asthelpers"
+	"github.com/mattermost/mattermost-server/plugin/checker/internal/version"
+
+	"github.com/pkg/errors"
+)
+
+func checkHelpersVersionComments(pkgPath string) (result, error) {
+	pkg, err := asthelpers.GetPackage(pkgPath)
+	if err != nil {
+		return result{}, err
+	}
+
+	api, apiIdent, err := asthelpers.FindInterfaceWithIdent("API", pkg.Syntax)
+	if err != nil {
+		return result{}, err
+	}
+
+	apiObj := pkg.TypesInfo.ObjectOf(apiIdent)
+	if apiObj == nil {
+		return result{}, errors.New("could not find type object for API interface")
+	}
+
+	helpers, err := asthelpers.FindInterface("Helpers", pkg.Syntax)
+	if err != nil {
+		return result{}, err
+	}
+
+	apiVersions := mapMinimumVersionsByMethodName(api.Methods.List)
+
+	helpersPositions := mapPositionsByMethodName(helpers.Methods.List)
+	helpersVersions := mapMinimumVersionsByMethodName(helpers.Methods.List)
+
+	implMethods := asthelpers.FindReceiverMethods("HelpersImpl", pkg.Syntax)
+	implVersions := mapEffectiveVersionByMethod(pkg.TypesInfo, apiObj.Type(), apiVersions, implMethods)
+
+	return validateMethods(pkg.Fset, helpersPositions, helpersVersions, implVersions), nil
+}
+
+func validateMethods(
+	fset *token.FileSet,
+	helpersPositions map[string]token.Pos,
+	helpersVersions map[string]version.V,
+	implVersions map[string]version.V,
+) result {
+	var res result
+
+	for name, helperVer := range helpersVersions {
+		pos := helpersPositions[name]
+
+		implVer, ok := implVersions[name]
+		if !ok {
+			res.Errors = append(res.Errors, renderWithFilePosition(
+				fset,
+				pos,
+				fmt.Sprintf("missing implementation for method %s", name)),
+			)
+			continue
+		}
+
+		if helperVer == "" {
+			res.Errors = append(res.Errors, renderWithFilePosition(
+				fset,
+				pos,
+				fmt.Sprintf("missing a minimum server version comment on method %s", name)),
+			)
+			continue
+		}
+
+		if helperVer == implVer {
+			continue
+		}
+
+		if helperVer.LessThan(implVer) {
+			res.Errors = append(res.Errors, renderWithFilePosition(
+				fset,
+				pos,
+				fmt.Sprintf("documented minimum server version too low on method %s", name)),
+			)
+		} else {
+			res.Warnings = append(res.Warnings, renderWithFilePosition(
+				fset,
+				pos,
+				fmt.Sprintf("documented minimum server version too high on method %s", name)),
```

Comment:
Is this something we want to enforce? (Just wondering if there may be a non-API reason why a function might be labelled higher than what is strictly required.)

Reply:
Yeah, my thoughts were that a helper might want to require a higher version in which a fix to an API was made, hence the warning. It's a purely theoretical concern right now.